crates/core/src/pattern/patterns.rs

```diff
@@ -539,14 +540,18 @@ impl Pattern {
                             )
                         })
                         .collect::<Result<Vec<Pattern>>>()?;
-                    // assumes that non multiple field has exactly one element
+
                     // TODO check if Pattern is Dots, and error at compile time,
                     // dots only makes sense in a list.
-                    if !field.multiple() {
-                        let lang = lang.get_ts_language();
-                        let field_name = lang.field_name_for_id(field_id).unwrap();
-                        let message = format!("field {} was empty!", field_name);
-                        return Ok((field_id, false, nodes_list.pop().expect(&message)));
+                    if !field.multiple() {    
+                        if nodes_list.len() == 1 {
+                            return Ok((field_id, false, nodes_list.pop().unwrap()));
+                        }
+                        let field_node = node.child_by_field_id(field_id).unwrap();
+                        let field_node_with_source = NodeWithSource::new(field_node.clone(), str::from_utf8(text).unwrap());
+                        return Ok((field_id, false,  Pattern::AstLeafNode(AstLeafNode::new(
+                            field_node.kind_id(), field_node_with_source.text(), lang,
+                        )?)));
```

Comment:
Reason for these changes?

Reply:
I also removed this comment
`// assumes that non multiple field has exactly one element`

I think the assumption that a node where multiple=false will have exactly 1 node is false, multiple=true means that you can have more than one element so this means multiple=false can have 0 (an unnamed node) or 1 node.

I added the code to handle the case where nodes_list is empty. such as binary_expression-> operator in php. (Where there are unnamed strings, such at "*" and "+" instead of named nodes.